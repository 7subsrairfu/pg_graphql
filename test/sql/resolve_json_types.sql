begin;

    create table exotic(
        id bigint primary key,
        js json not null,
        jsb jsonb not null
    );

    insert into exotic(id, js, jsb)
    values (1, '{"hello": 1}', '["world", "other"]');

    select jsonb_pretty(
        graphql.resolve($$
            {
              exoticCollection {
                edges {
                  node {
                    id
                    js
                    jsb
                  }
                }
              }
            }
        $$)
    );

rollback;
