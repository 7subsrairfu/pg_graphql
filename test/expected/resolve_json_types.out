begin;
    create table exotic(
        id bigint primary key,
        js json not null,
        jsb jsonb not null
    );
    insert into exotic(id, js, jsb)
    values (1, '{"hello": 1}', '["world", "other"]');
    select jsonb_pretty(
        graphql.resolve($$
            {
              exoticCollection {
                edges {
                  node {
                    id
                    js
                    jsb
                  }
                }
              }
            }
        $$)
    );
              jsonb_pretty              
----------------------------------------
 {                                     +
     "data": {                         +
         "exoticCollection": {         +
             "edges": [                +
                 {                     +
                     "node": {         +
                         "id": "1",    +
                         "js": {       +
                             "hello": 1+
                         },            +
                         "jsb": [      +
                             "world",  +
                             "other"   +
                         ]             +
                     }                 +
                 }                     +
             ]                         +
         }                             +
     },                                +
     "errors": [                       +
     ]                                 +
 }
(1 row)

rollback;
